src/sdl2/video.rs

```diff
@@ -449,11 +450,33 @@ impl GLContext {
     }
 }
 
-pub struct Window {
+/// Holds a `SDL_Window`
+///
+/// When the `WindowContext` is dropped, it destroys the `SDL_Window`
+///
+/// *INTERNAL USE ONLY*
+pub struct WindowContext {
     subsystem: VideoSubsystem,
     raw: *mut ll::SDL_Window,
 }
 
+impl Drop for WindowContext {
+    #[inline]
+    fn drop(&mut self) {
+        unsafe { ll::SDL_DestroyWindow(self.raw) };
+    }
+}
+
+/// Holds a `Rc<WindowContext>`.
+///
+/// Note: If a `Window` goes out of scope but it cloned its context,
+/// then the `SDL_Window` will not be destroyed until there are no more references to the `WindowContext`.
+/// This may happen when a `Canvas` that is created from a `Window` creates a `TextureCreator`.
```

Comment:
You can shorten this to "This may happen when a `TextureCreator<Window>` outlives its `Canvas<Window>`", it's less verbose, more concise and it's straight to the point, users don't really care about how it is implemented.